pkg/controller/directvolumemigration/rsync.go

```diff
@@ -0,0 +1,827 @@
+		} else if err != nil {
+			return err
+		}
+		destSecret := corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: ns,
+				Name:      "directvolumemigration-rsync-creds",
+				Labels: map[string]string{
+					"app": "directvolumemigration-rsync-transfer",
+				},
+			},
+			Data: map[string][]byte{
+				"credentials": []byte("root:changeme"),
+			},
+		}
+		err = destClient.Create(context.TODO(), &destSecret)
+		if k8serror.IsAlreadyExists(err) {
+			t.Log.Info("Secret already exists on destination", "namespace", destSecret.Namespace)
+		} else if err != nil {
+			return err
+		}
+	}
+
+	// One rsync transfer pod per namespace
+	// One rsync client pod per PVC
+
+	// Also in this rsyncd configmap, include all PVC mount paths, see:
+	// https://github.com/konveyor/pvc-migrate/blob/master/3_run_rsync/templates/rsyncd.yml.j2#L23
+
+	return nil
+}
+
+// Create rsync transfer route
+func (t *Task) createRsyncTransferRoute() error {
+	// Get client for destination
+	destClient, err := t.getDestinationClient()
+	if err != nil {
+		return err
+	}
+	pvcMap := t.getPVCNamespaceMap()
+	for ns, _ := range pvcMap {
+		svc := corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "directvolumemigration-rsync-transfer-svc",
+				Namespace: ns,
+				Labels: map[string]string{
+					"app": "directvolumemigration-rsync-transfer",
+				},
+			},
+			Spec: corev1.ServiceSpec{
+				Ports: []corev1.ServicePort{
+					{
+						Name:       "stunnel",
+						Protocol:   corev1.ProtocolTCP,
+						Port:       int32(2222),
+						TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: 2222},
+					},
+				},
+				Selector: map[string]string{
+					"app":     "directvolumemigration-rsync-transfer",
+					"purpose": "rsync",
+					"owner":   "directvolumemigration",
+				},
+				Type: corev1.ServiceTypeClusterIP,
+			},
+		}
+		err = destClient.Create(context.TODO(), &svc)
+		if k8serror.IsAlreadyExists(err) {
+			t.Log.Info("Rsync transfer svc already exists on destination", "namespace", ns)
+		} else if err != nil {
+			return err
+		}
+		route := routev1.Route{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "directvolumemigration-rsync-transfer-route",
+				Namespace: ns,
+				Labels: map[string]string{
+					"app": "directvolumemigration-rsync-transfer",
+				},
+			},
+			Spec: routev1.RouteSpec{
+				To: routev1.RouteTargetReference{
+					Kind: "Service",
+					Name: "directvolumemigration-rsync-transfer-svc",
+				},
+				Port: &routev1.RoutePort{
+					TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: 2222},
+				},
+				TLS: &routev1.TLSConfig{
+					Termination: routev1.TLSTerminationPassthrough,
+				},
+			},
+		}
+		err = destClient.Create(context.TODO(), &route)
+		if k8serror.IsAlreadyExists(err) {
+			t.Log.Info("Rsync transfer route already exists on destination", "namespace", ns)
+		} else if err != nil {
+			return err
+		}
+		t.RsyncRoutes[ns] = route.Spec.Host
```

Comment:
also, we do certain validations in pvc-migrate to indicate whether a route is admitted in the system or not. Probably need to add that check(may in the same phase I am not really sure)